pymeasure/instruments/temptronic/temptronic_base.py

```diff
@@ -0,0 +1,826 @@
+
+        temp_limit_air_low: float
+            minimum flow temperature limit
+
+        temp_limit_air_high: float
+            maximum flow temperature limit
+
+        temp_limit_air_dut: float
+            maximum allowed difference between DUT and Flow temperature
+
+        Returns
+        -------
+        none
+
+        """
+        self.maximum_test_time = 3600*5   # max 5 houre test time
+        #sent dut type
+        self.dut_type=dut_type
+        #temperature window
+        self.temperature_setpoint_window = temp_window
+        #temperature limit low
+        self.temperature_limit_air_low = temp_limit_air_low
+        #temperature limit high
+        self.temperature_limit_air_high = temp_limit_air_high
+        self.dut_mode = 'ON'
+        self.dut_constant = dut_constant
+        self.temperature_limit_air_dut = temp_limit_air_dut
+        self.temperature_soak_time = soak_time
+
+    def set_temperature(self, set_temp):
+        """sweep to a specified.
+
+        Parameters
+        ----------
+        set_temp : float
+            target temperature for DUT
+
+        Returns
+        -------
+        none
+
+        """
+        if self.mode == 'manual':
+            if set_temp <= 20:
+                self.set_point_number = 2 # cold
+            elif set_temp < 30:
+                self.set_point_number = 1 # ambient
+            elif set_temp >= 30:
+                self.set_point_number = 0 # hot
+            else:
+                raise ValueError(f"Temperature {set_temp} is impossible to set!")
+
+        self.temp_set_point=set_temp
+
+    def wait_for_settling(self, time_limit=300, use_logging=False):
+        """sweep to a specified.
+
+        Parameters
+        ----------
+        set_temp : float
+            target temperature for DUT
+
+        time_limit : float
+            set the maximum blocking time within thermostream has to settle.
+            Script execution is blocked until either thermostream has settled
+            or time_limte has been exceeded.
+
+        use_logging : boolean
+            show polling progress in command line
+
+        Returns
+        -------
+        none
+
+        """
+
+        time.sleep(1)
+        t = 0
+        while( not self.at_temperature()): #assert at temperature
+            time.sleep(1)
+            t += 1
+            if use_logging == True:
+                log.info(f"temp_set= {self.temperature_setpoint:4.1f} deg,",
+                         f"temp = {self.temperature:4.1f} deg,",
+                         f"time elapsed = {t:4.0f} s,",
+                         f"status = {self.temperature_event_status}")
+
+            if t > time_limit:
+                log.info('no settling achieved')
+                break
+
+    def shutdown(self):
+        """Brings the instrument to a safe and stable state"""
+        super().isShutdown = True
+        log.info("Shutting down %s" % self.name)
```

Comment:
These lines duplicate what is currently in `Instrument.shutdown()`. Better to do stuff specific to your instrument (the next 3 lines) first, then call ` super().shutdown()`. That should do the trick.

Reply:
Incorporated like suggested :)